lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3688,6 +3704,64 @@ void TFFunctionPartition::balanceRetainReleaseCount(SILValue oldResult,
   }
 }
 
+void TFFunctionPartition::insertReplacementGraphOp(
+    ArrayRef<SILValue> resultValues) {
+  // Sanity check that all result values are tensor handles. Note SIL
+  // accelerator functions under the "tensorflow" convention can also return
+  // variant and resource tensors (in addition to tensor handles), but such
+  // functions will not generate all host-side code, and thus this method will
+  // not be called.
+  assert(TFSendRecvOpaqueHandle ||
+         std::all_of(resultValues.begin(), resultValues.end(),
+                     [](SILValue resultValue) {
+                       return isTensorHandle(resultValue->getType());
+                     }) &&
+             "Cannot return a non-TensorHandle value to host in the TF program "
+             "-- should this function use tensorflow convention?");
+
+  auto &ctx = hostFn.getASTContext();
+  auto loc = hostFn.getLocation();
+
+  SILBuilder B(tensorStartPoint);
+
+  SmallVector<SILType, 4> ResultSILTypes;
+  for (auto &resultValue : resultValues) {
+    ResultSILTypes.push_back(resultValue->getType());
+  }
+
+  std::string suffix;
+  for (size_t i = 0; i < tensorFnArguments.size(); ++i) {
+    suffix += ",i";
```

Comment:
if we use a GraphOperationBuilder, i guess we don't need to do such mangling. Would that be better? You can assess.